src/main.rs

```diff
@@ -139,27 +141,37 @@ fn main() -> anyhow::Result<()> {
                 notify::event::DataChange::Content,
             )) = event.kind
             {
-                let shape = match model.load(&parameters) {
-                    Ok(shape) => shape,
-                    Err(model::Error::Compile) => {
-                        // It would be better to display an error in the UI,
-                        // where the user can actually see it. Issue:
-                        // https://github.com/hannobraun/fornjot/issues/30
-                        println!("Error compiling model");
-                        return;
-                    }
-                    Err(err) => {
-                        panic!("Error reloading model: {:?}", err);
-                    }
-                };
-
-                // This will panic, if the other end is disconnected, which is
-                // probably the result of a panic on that thread, or the
-                // application is being shut down.
-                //
-                // Either way, not much we can do about it here, except maybe to
-                // provide a better error message in the future.
-                watcher_tx.send(shape).unwrap();
+                let file_ext = event
+                    .paths
+                    .get(0)
+                    .expect("File path missing in watch event")
+                    .extension();
+
+                let black_list = HashSet::from([
+                    OsStr::new("swp"),
+                    OsStr::new("tmp"),
+                    OsStr::new("swx"),
+                ]);
+
+                if (file_ext.is_some()
+                    && black_list.get(file_ext.unwrap()).is_none())
+                    | file_ext.is_none()
```

Comment:
This is just a nit-pick. Feel free to act on it (or comment, if you disagree), or just leave it as-is, as you prefer. It doesn't matter a whole lot.

That `file_ext.is_some()` followed by `file_ext.unwrap()` is a bit icky. `if let Some(ext) = file_ext` would be more robust, but can't be used as a direct replacement here.

Alternative approach:
``` rust
if let Some(ext) = file_ext {
    if black_list.contains(ext) {
        return;
    }
}

// insert rest of event handling here
```

Aside from using the more robust `if let`, this also makes the handling of the error case a self-contained thing, thereby making the code a bit easier to understand. It also saves one level of indentation for the happy path. But as I said, not that important, feel free to ignore.

Reply:
Thanks for the feedback.

Removing the comment was an accident - sorry for that.

The alternative approach indeed looks neater. Still getting used to the rust coding style coming from C/C++.
I've pushed the changes

Reply:
Hey, no problem, that what reviews are for! :smile: